WordPress/Classes/Models/Blog.m

```diff
@@ -292,9 +293,34 @@ - (NSString *)postFormatTextFromSlug:(NSString *)postFormatSlug
 // WP.COM private blog.
 - (BOOL)isPrivate
 {
-    return (self.isHostedAtWPcom && [[self getOptionValue:@"blog_public"] isEqualToString:@"-1"]);
+    return (self.isHostedAtWPcom && [self.privacy isEqualToNumber:@(BlogPrivacyPrivate)]);
```

Comment:
**Question:** are we migrating the legacy options, somehow, to the new `privacy` field? (OR refreshing it in a sync call?). Otherwise, right after the migration, `self.privacy` will get the default value, and (everything) that relies on it will fail.